docs/protocols.ipynb

```diff
@@ -181,16 +145,112 @@
     "id": "b4d4bc702a5e"
    },
    "source": [
-    "## What is a Protocol\n",
+    "## What is a protocol? \n",
+    "\n",
+    "A protocol is a combination of the following two items: \n",
+    "- a `SupportsXYZ` class, which defines and documents all the magic functions that need to be implemented in order to support that given protocol \n",
+    "- the entrypoint function(s), which are exposed to the main cirq namespace as `cirq.xyz()`\n",
     "\n",
-    "TODO: describe the class SupportsXYZ, private methods, the entry-point cirq.xyz and the fallback strategies.\n",
+    "Note: While the protcol is technically both of these things, we refer to the public utility functions interchangeably as protocols. See the list of them below.\n",
     "\n",
     "\n",
     "## Cirq's protocols\n",
     "\n",
-    "TODO: draw a nice diagram of what areas the current protocols belong to and how they are interrelated\n",
+    "Cirq's protocols can be useful in multiple feature areas of quantum computing. Here's a list of all of them. \n",
+    "\n",
+    "| Protocol | Areas | \n",
+    "|----------|-------|\n",
+    "|`cirq.act_on`| simulation |\n",
```

Comment:
Should we add a short description of each protocol?  

Also, these areas are a bit fuzzy and awkward.  I use cirq.unitary for analysis, for instance.  I guess it is analysis of a simulation, but that means that "simulation" covers a broad ground.  Also, this is referred to as "Quantum operator representation" in the next section, which is not really simulation.

I wonder if it would just be better to have a short description of the function and people can use them how they want.